engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/selectModulesScreen/SelectModulesScreen.java

```diff
@@ -359,46 +362,89 @@ public String get() {
                         .filter(info -> info.isSelected() && info.isExplicitSelection()).forEach(this::deselect));
             }
 
-            localOnlyCheckbox = find("localOnly", UICheckbox.class);
-            localOnlyCheckbox.bindChecked(
-                    new Binding<Boolean>() {
-
-                        @Override
-                        public Boolean get() {
-                            filterText(moduleSearch.getText());
-                            prepareModuleList(selectModulesConfig.isChecked());
-                            return selectModulesConfig.isChecked();
-                        }
+			for (CheckboxAssociation checkboxAssociation : CheckboxAssociation.values()) {
+				String checkboxName = checkboxAssociation.getCheckboxName();
+				StandardModuleExtension standardModuleExtension = checkboxAssociation.getStandardModuleExtension();
+
+				UICheckbox checkBox = find(checkboxName, UICheckbox.class);
+				if (null != checkBox) {
+					checkBox.setChecked(selectModulesConfig.isStandardModuleExtensionSelected(standardModuleExtension));
+					checkBox.subscribe(e -> {
+						selectModulesConfig.toggleStandardModuleExtensionSelected(standardModuleExtension);
+						checkBox.setChecked(selectModulesConfig.isStandardModuleExtensionSelected(standardModuleExtension));
+						filterModules();
+					});
+				} else {
+					logger.error("Unable to find checkbox named " + checkboxName + " in " + ASSET_URI.toString());
+					selectModulesConfig.unselectStandardModuleExtension(standardModuleExtension);
+				}
+			}
+
+			UICheckbox localOnlyCheckbox = find("localOnlyCheckbox", UICheckbox.class);
+			localOnlyCheckbox.setChecked(selectModulesConfig.isLocalOnlyChecked());
+			localOnlyCheckbox.subscribe(e -> {
+				selectModulesConfig.setIsLocalOnlyChecked(!selectModulesConfig.isLocalOnlyChecked());
+				localOnlyCheckbox.setChecked(selectModulesConfig.isLocalOnlyChecked());
+				filterModules();
+			});
+        }
+        WidgetUtil.trySubscribe(this, "close", button -> triggerBackAnimation());
+    }
 
-                        @Override
-                        public void set(Boolean value) {
-                            selectModulesConfig.setIsChecked(value);
-                            filterText(moduleSearch.getText());
-                            prepareModuleList(value);
-                        }
-                    }
-            );
+     private void filterModules() {
+        sortedModules.clear();
+        sortedModules.addAll(allSortedModules);
+        if(selectModulesConfig.isAnyStandardModuleExtensionSelected()) {
```

Comment:
This will always happen if there's a Boolean StandardModuleExtension that isn't in the checkbox association table.   Another reason to get rid of the table.